### ISSUE TITLE
[MMdetection 3]  How should I use gradient accumulation correctly

### ISSUE THREAD

**Issue by mf991** [open-mmlab/mmdetection#10728]
In mmdetection, Ir is 0.01 when batch-size is 8*2=16.
Due to hardware problems, I can only set the batch to 8 on a single gpu. So I want to use gradient accumulation to solve this problem.

The following is part of the config file：
train_dataloader = dict(
    batch_size=8,
    ......）
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001))  
    accumulative_counts=2)  # Add this line to set the batch to 16.
auto_scale_lr = dict(enable=True, base_batch_size=16)

Now there is a question, should my learning rate be set to 0.005（0.01/2） or 0.01？（After all, the actual batch-size is still 8）
I don't quite understand the internals of gradient accumulation.  And how to set the learning rate to make it closest to batch_size=16, Ir=0.01 on a single gpu?

**Comment by wzwcs** [comment]
I miss the same question, have you solved it?

**Comment by sidgairo18** [comment]
For your question - 
Ir is 0.01 when batch-size is 8*2=16. 

Now when running on single-gpu with batch-size=8 and accumulative_counts=2, then you will use the same lr = 0.01 as the batch-size is 8*2=16 (or you can assume it to be the effective batch-size).
